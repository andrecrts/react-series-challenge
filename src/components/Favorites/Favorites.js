import React, { Component } from 'react'
import { connect } from 'react-redux'
import contentActions from '../../actions/content'
import Header from '../Header'
import Searchbar from '../Searchbar'
import Item from '../Item'

const {
  addToFavContent,
  removeFavContent,
} = contentActions.actions

const Content = ({data, onClickFav, favorites, filter}) => {
  const loading = !data
  if(loading){
    return(
      <div> loading ... </div>
    )
  }

  return Object.keys(data).map((key, index) => {
    if(data[key].title.search(filter) > -1){
      return(
        <Item key={index}
          item={data[key]}
          onClickFav={onClickFav}
          isFav={favorites[key] ? true : false}
        />
      )
    }
    return null
  })

}

class Favorites extends Component {

  constructor(props) {
    super(props)
    this.toogleFavorite = this.toogleFavorite.bind(this)
    this.state = {
      searchTerm: ''
    }
  }

  toogleFavorite(item, isFav) {
    if(!isFav){
      this.props.addToFavContent(item)
    }
    else {
      this.props.removeFavContent(item)
    }
  }

  handleKeyPress = (e) => {
    if (e.key === 'Enter') {
      this.setState({searchTerm: e.target.value})
    }
  }

  render() {
    return(
      <div>
        <Header/>
        <Searchbar
          handleKeyPress={this.handleKeyPress}
        />
        <Content
          data={this.props.content.favorites}
          onClickFav={this.toogleFavorite}
          favorites={this.props.content.favorites}
          filter={this.state.searchTerm}
        />
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    content: state.content
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    addToFavContent: (item) => {
      dispatch(addToFavContent(item))
    },
    removeFavContent: (item) => {
      dispatch(removeFavContent(item))
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Favorites)

