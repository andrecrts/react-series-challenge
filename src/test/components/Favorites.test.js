import React from 'react';
import Favorites from '../components/Favorites'
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'

const middlewares = []
const mockStore = configureStore(middlewares)

test('Home snapshot', () => {
  const initialState = {
    content:{
      data: {data: {tite:'title1', images:{original:{webp:'srcImage'}}}},
      favorites:{}
    }
  }
  const store = mockStore(initialState)
  const component = renderer.create(
    <Favorites store={store}/>,
  );
  let componentJSON = component.toJSON();
  expect(componentJSON).toMatchSnapshot();
})

