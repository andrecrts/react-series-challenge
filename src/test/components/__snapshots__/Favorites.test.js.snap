// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot 1`] = `
<div>
  <div
    className="sc-bdVaJa cpMCnx"
  >
    <input
      className="sc-bwzfXH iYnflE"
      onChange={[Function]}
      onKeyPress={null}
      placeholder="What are you looking for..."
      type="text"
    />
  </div>
  <div
    className="sc-bZQynM loJkTV"
  >
    <div>
       No Favorites 
    </div>
  </div>
</div>
`;
